cpp/src/utilities/cuda_utils.cuh

```diff
@@ -0,0 +1,43 @@
+/*
```

Comment:
Another header with cuda utils functions?
We may want to to reconcile with `graph_utils` and `sm_utils`

Reply:
This was necessary just for bringing in the newest cuML changes to the weak_cc implementation, so I just added it here for simplicity.

We could certainly consolidate it with some of the other existing files.  I was imagining we might want to do a more comprehensive effort at that at some point, but I could take a shot at that with this PR if we like.